src/lib/components/references/bibleTextReferenceFetcher.ts

```diff
@@ -20,191 +20,108 @@ export const fetchAndFormat = async (
-                                startVerseMapping!.targetVerses[0]!.verseId
-                            } - EndVerseId: ${startVerseMapping!.targetVerses.at(-1)!.verseId} - Error: ${error}`
-                        )
-                    );
-                    return null;
-                }
-
-                // now, if we do not have content, we should return null and throw error
-                if (passageBookTexts.length === 0) {
-                    logUnexpectedError(bookTexts, startVerse, endVerse, language);
-                    return null;
-                }
-            }
-
-            verseDisplayName = generateVerseReference(passageBookTexts, language, true);
-        } else {
-            // multiple verse passage with at least some mappings
-            // ! any given VerseMapping.targetVerse in the mapping can be null.
-            // ! This represents an exclusion = verse does not exist in target bible
-
-            // check for mapping for start and end verses
-            const startVerseMapping = findMappingByVerseId(startVerse, versificationMappings);
-            const endVerseMapping = findMappingByVerseId(endVerse, versificationMappings);
-            let passageStartVerseId = startVerse;
-            let passageEndVerseId = endVerse;
-
-            if (startVerseMapping) {
-                if (startVerseMapping.targetVerses === null || startVerseMapping.targetVerses.length === 0) {
-                    // exclusion - this passage, or part of it, does not exist in target bible
-                    return null;
-                } else {
-                    passageStartVerseId = startVerseMapping.targetVerses[0]!.verseId;
-                }
-            }
-
-            if (endVerseMapping) {
-                if (endVerseMapping.targetVerses === null || endVerseMapping.targetVerses.length === 0) {
-                    // exclusion - this passage, or part of it, does not exist in target bible
-                    return null;
-                } else {
-                    passageEndVerseId = endVerseMapping.targetVerses.at(-1)!.verseId;
-                }
+        if (endVerseMapping) {
+            if (endVerseMapping.targetVerses === null || endVerseMapping.targetVerses.length === 0) {
+                // exclusion - this passage, or part of it, does not exist in target bible
+                return null;
+            } else {
+                targetBibleEndVerseId = endVerseMapping.targetVerses.at(-1)!.verseId;
             }
+        }
 
-            // does passage exist in current bookTexts?
-            passageBookTexts = getMultiVersePassageContentFromBookTexts(
-                bookTexts,
-                parseVerseId(passageStartVerseId),
-                parseVerseId(passageEndVerseId)
+        // handle unexpected versification mapping data
+        if (targetBibleStartVerseId > targetBibleEndVerseId) {
+            log.exception(
```

Comment:
This exception is new and seems worthy of logging so we can look into our data to see what's going on.